lib/rexml/parsers/baseparser.rb

```diff
@@ -738,6 +721,18 @@ def parse_id_invalid_details(accept_external_id:,
         end
       end
 
+      def process_comment
+        text = @source.read_until("-->")
+        if text.chomp!("-->")
+          if text.include? "--" or text.end_with?("-")
+            raise REXML::ParseException.new("Malformed comment", @source)
+          end
+          return text
+        else
+          raise REXML::ParseException.new("Unclosed comment: Missing end '-->'", @source)
+        end
```

Comment:
How about using early return to avoid a nest?

```suggestion
        unless text.chomp!("-->")
          raise REXML::ParseException.new("Unclosed comment: Missing end '-->'", @source)
        end

        if text.include? "--" or text.end_with?("-")
          raise REXML::ParseException.new("Malformed comment", @source)
        end
        text
```

Reply:
OK
I see.